src/SecretManager/Microsoft.DncEng.SecretManager/SecretManagerCredentialProvider.cs

```
@@ -8,16 +11,40 @@ namespace Microsoft.DncEng.SecretManager;
 
 public sealed class SecretManagerCredentialProvider : ITokenCredentialProvider
 {
+    /// <inheritdoc/>
+    public string ApplicationId { get; internal set; }
+    /// <inheritdoc/>
+    public string TenantId { get; internal set; }
+
+    public SecretManagerCredentialProvider()
+    {
+        SetCredentialIdentityValues();
+    }
+
     // Expect AzureCliCredential for CI and local dev environments. 
     // Use InteractiveBrowserCredential as a fallback for local dev environments.
     private readonly Lazy<TokenCredential> _credential = new(() =>
         new ChainedTokenCredential(
             new AzureCliCredential(new AzureCliCredentialOptions { TenantId = ConfigurationConstants.MsftAdTenantId }),
             new InteractiveBrowserCredential(new InteractiveBrowserCredentialOptions() { TenantId = ConfigurationConstants.MsftAdTenantId })
         ));
-
+        
     public Task<TokenCredential> GetCredentialAsync()
     {
         return Task.FromResult(_credential.Value);
     }
+
+    /// <inheritdoc/>
+    internal void SetCredentialIdentityValues()
+    {
+        // Get a token from the crendential provider
+        var tokenRequestContext = new TokenRequestContext(new[] { "https://management.azure.com/.default" });
+        var token = _credential.Value.GetToken(tokenRequestContext, CancellationToken.None);
+
+        // Decode the JWT to get user identity information
+        var handler = new JwtSecurityTokenHandler();
+        var jsonToken = handler.ReadToken(token.Token) as JwtSecurityToken;
+        ApplicationId = jsonToken?.Claims?.FirstOrDefault(claim => claim.Type == "appid")?.Value ?? "Claim appid (Application Id) Not Found"; 
+        TenantId = jsonToken?.Claims?.FirstOrDefault(claim => claim.Type == "tid")?.Value ?? "Claim tid (Tenant Id) Not Found";
```

should these expressions `throw` instead of setting the properties to invalid values❓

---

Similar to the other comment I mad this method non-volatile. We don't want this process to kill the service if it fails and we don't want to lose logging even these values are not found. The logs still have value even if not all identity values are found.

---

Based on the latest security advisor responses this logging will always occur in a non-volatile way. Meaning we always log with as much valid information as possible and any values we cannot extract are set to some default value.